### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by mvpatel2000**
Remove deprecated code related to grad_accum and old precision names

Repoint LLM code

### CODE REVIEW

File: examples/common/config_utils.py [hunk 1]
```diff
@@ -9,39 +9,8 @@
 from omegaconf import OmegaConf as om
 
 
-def calculate_batch_size_info(global_batch_size: int,
```
Comment (dakinggg) [inline examples/common/config_utils.py:12]: the error message in this function still seems useful?

**Comment by mvpatel2000** [comment]
Oops... still need to be bump... it breaks a few tests...

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -18,6 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 20
     strategy:
+      fail-fast: false  # TODO: Delete
```
Comment (dakinggg) [inline .github/workflows/lint.yaml:21]: delete?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -18,6 +18,7 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 20
     strategy:
+      fail-fast: false  # TODO: Delete
```
Comment (dakinggg) [inline .github/workflows/test.yaml:21]: delete?

File: examples/bert/tests/test_main.py [hunk 4]
```diff
@@ -22,9 +22,9 @@ def test_trainer(model_name: str, seed: int):
 
     with SynthTextDirectory() as tmp_datadir:
         config.train_loader.dataset.remote = tmp_datadir
-        config.train_loader.dataset.local = tmp_datadir
+        config.train_loader.local = os.path.join(tmp_datadir, 'local2')
```
Comment (dakinggg) [inline examples/bert/tests/test_main.py:25]: i don't think removing the `dataset` part of the prefix here is correct?

File: examples/bert/tests/test_main.py [hunk 5]
```diff
@@ -22,9 +22,9 @@ def test_trainer(model_name: str, seed: int):
...
         config.eval_loader.dataset.remote = tmp_datadir
-        config.eval_loader.dataset.local = tmp_datadir
+        config.eval_loader.local = os.path.join(tmp_datadir, 'local2')
```
Comment (dakinggg) [inline examples/bert/tests/test_main.py:27]: same

File: examples/bert/tests/test_main.py [hunk 6]
```diff
@@ -40,6 +40,8 @@ def test_trainer(model_name: str, seed: int):
         # Check that the checkpoint was loaded by comparing model weights (with no weight changes)
         config.load_path = chkpt_path
         config.seed += 10  # change seed
+        config.train_loader.local = os.path.join(tmp_datadir, 'local2')
```
Comment (dakinggg) [inline examples/bert/tests/test_main.py:43]: same

File: requirements.txt [hunk 7]
```diff
@@ -2,7 +2,7 @@
 # Each example subdir has its own `requirements.txt` with extra deps.
 # E.g. to install LLM folder deps, use `pip install -e .[llm]`
 
-mosaicml>=0.13.4,<0.14
+mosaicml>=0.13.4,<0.15
```
Comment (dakinggg) [inline requirements.txt:5]: should this be bumped to 0.14 min?

**Review by dakinggg — APPROVED** [review]

File: requirements.txt (hunk 7 above)
Comment (mvpatel2000) [reply requirements.txt:5]: llm is pinned to lower version still (not bothering to upgrade since we'll remove) so this is looser

**Pull request merged**